### ISSUE TITLE
Front-end highlighter redesign: Missing heading markup

### ISSUE THREAD

**Issue by amberhinds** [equalizedigital/accessibility-checker#1619]
There are no headings in the front-end highlighter dialog, though visually text is styled as headings. 

## Example code:
```
<div class="edac-highlight-panel-controls-title"><span class="edac-highlight-panel-controls-title-icon" aria-hidden="true"></span>Accessibility Checker</div>

<span class="edac-highlight-panel-description-title-text">Possible Heading</span>
```
## Screenshots

<img width="1168" height="497" alt="Image" src="https://github.com/user-attachments/assets/a3f218dd-5351-4103-85a2-20f845d76b4c" />

<img width="1147" height="584" alt="Image" src="https://github.com/user-attachments/assets/013be75c-3f7c-4425-bb42-2ea9846a0b7b" />

<span class="edac-highlight-summary-total">17 issues found</span>

## Expected heading levels:
- Dialog title: H2
- Description/Issue title: H3
- WCAG: H4
- Severity: H4
- Why it matters: H4
- How to fix: H4
- Summary total: H3

If style conflicts are a concern, use `role="heading"` and `aria-level` to define headings for screen reader users (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/heading_role)

**Comment by pattonwebz** [comment]
This was tackled in #1621. I added the role as swapping the elements caused a lot of style issues since we do an `unset: all`
